docs/the-algorithm.rst

```diff
@@ -0,0 +1,132 @@
+The algorithm
+=============
+
+What follows is a walkthrough, with visual aids, of the algorithm underlying this package.
+
+Background
+~~~~~~~~~~
+
+Let's start with a simple GeoJSON geometry that doesn't come anywhere near the antimeridian.
+For the sake of example, let's say it runs from 40° longitude west to 40° east, and from 20° latitude south to 20° north.
+It's lower left point is at ``-40, -20``, and its upper right point is at ``40, 20``.
+
+.. figure:: img/box.png
+    :scale: 30%
+    :align: center
+
+    A simple box.
+
+Now let's consider the same geometry, but shifted east (or west) 180 degrees.
+Its lower-left point is ``140, -20``, and its upper-right point is at ``220, 20``.
+
+.. figure:: img/crossing.png
+    :scale: 30%
+    :align: center
+
+    A box that crosses the antimeridian.
+
+So far, good enough.
+However, the bounds of WGS84, the coordinate system used by GeoJSON, go from -180° to 180° longitude.
+This means that the upper-right coordinate of our antimeridian-crossing box are often "wrapped" to ``-140, 20``.
+This leads to this confusing visual representation.
+
+.. figure:: img/split.png
+    :scale: 30%
+    :align: center
+
+    A box that crosses the antimeridian, wrapped.
+
+Let's consider the four points of our box, and the order in which they're connected.
+Exterior rings in GeoJSON should be wound counter-clockwise, but you can see that our antimeridian-crossing box appears to be wound clockwise.
+
+.. figure:: img/winding.png
+    :scale: 30%
+    :align: center
+
+    The four points in our box, with their connections.
+
+This leads to a confusion situation, where the "correct" interpretation of our antimeridian-crossing box would be globe-covering shape with a "hole" over the equator and the prime meridian.
+This is very rarely what you're actually trying to represent.
+
+.. figure:: img/winding-repercussions.png
+    :scale: 30%
+    :align: center
+
+    If we interpret our antimeridian-crossing box literally, it describes a very strange geometry.
+
+Here's where our algorithm comes in.
+
+Segmentation
+~~~~~~~~~~~~
+
+Given our antimeridian-crossing box as our input, we walk through the points until we find a longitude jump of more than 180° degrees.
+
+.. figure:: img/delta-gt-180.png
+    :scale: 30%
+    :align: center
+
+    A longitude jump greater than 180°.
+
+When we find such a jump, we split the polygon into segments by inserting two points, one on each antimeridian.
```

Comment:
How do we know the direction of the split segments? It's opposite the direction of the segment that exceeds 180 degrees.

Reply:
This sort of takes care of itself, from the algorithm's perspective. One of the antimeridian points forms the end of the "closed" segment, and the other antimeridian point is the start of the next segment.